shanoir-ng-datasets/src/main/java/org/shanoir/ng/dicom/web/StudyInstanceUIDHandler.java

```diff
@@ -0,0 +1,174 @@
+package org.shanoir.ng.dicom.web;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.annotation.PostConstruct;
+
+import org.shanoir.ng.dataset.model.Dataset;
+import org.shanoir.ng.dataset.model.DatasetExpression;
+import org.shanoir.ng.dataset.model.DatasetExpressionFormat;
+import org.shanoir.ng.datasetacquisition.model.DatasetAcquisition;
+import org.shanoir.ng.datasetfile.DatasetFile;
+import org.shanoir.ng.examination.model.Examination;
+import org.shanoir.ng.examination.service.ExaminationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+
+@Component
+public class StudyInstanceUIDHandler {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(StudyInstanceUIDHandler.class);
+
+	private static final String WADO_URI_STUDY_UID_SERIES_UID = "studyUID=(.*?)\\&seriesUID";
+	
+	private static final String WADO_RS_STUDY_UID_SERIES_UID = "/studies/(.*?)/series/";
+
+	@Autowired
+	private ExaminationService examinationService;
```

Comment:
I can see no method where this cache is evicted/updated.
If a user deletes/updates an examination (or the data behind it) it should be updated/evicted no ?

Reply:
- if an user deletes an examination, no display of images can be called on this exam via the GUI
- for me, not an issue if the cache until the night would provide an UID, that is never called
- update exam, does not change the dataset files normally, even if deleted, but the UID remains unique for all series of one exam

Reply:
Hi Jean-Come, would you be okay to close this conversation? I think there is really no value from cleaning the cache, except cleaning it via a scheduler e.g. every 24 hours or once a week. It only introduces more code complexity. Our prod and it will be the same for the GME is stopped for security updates all night, so the cache is cleaned in the morning. The info in the cache is so small, even if during one morning somebody we read 500 exams, nothing would impact that on the server. Exams are in practice never deleted once imported. And nobody except typing the URL can open a deleted exam into the viewer and examIds are counted up all the time. The viewer would just throw an error.

Reply:
If the cache is evicted every 24h it's OK for me. let's keep this in mind if we have a bug with the OHIF viewer one of these day ;)
I'm closing it!